packages/language-server/src/server.ts

```diff
@@ -407,6 +411,12 @@ connection.onWorkspaceSymbol(({ query }) => {
 })
 
 connection.onDocumentFormatting(async ({ textDocument: { uri }, options }) => {
+	const errors = service.project.cacheService.errors[uri]
+	if (errors && errors.some(error => error.severity === core.ErrorSeverity.Error)) {
+		// Don't format if there are errors.
```

Comment:
Instead of checking the `cacheService`, it seems better to me to check the `parseErrors` on the document's AST file node. Checker and linter errors shouldn't prevent a file from being formatted, and I also don't know whether the cacheService errors always fully match the AST.